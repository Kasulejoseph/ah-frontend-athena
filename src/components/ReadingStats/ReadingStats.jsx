import React from "react";
import propTypes from "prop-types";
import {
  MDBCard,
  MDBCardBody,
  MDBCardText,
  MDBCol,
  MDBRow,
  MDBContainer
} from "mdbreact";
import "./ReadingStats.scss";

const ReadingStats = props => {
  const { article } = props;
  return (
    <MDBContainer>
      <MDBCol className="flex flex-center">
        <MDBCard className="w-50">
          <MDBCardBody>
            <MDBCardText>
              <MDBRow>
                <MDBCol size="8">
                  <h3 className="font-weight-bold mb-3 p-0">{article.title}</h3>
                </MDBCol>
                <MDBCol size="2">
                  <h1 className="font-weight-bold mb-3 p-0 black-text">
                    {article.read_count}
                  </h1>
                  <span className="black-text">reads</span>
                </MDBCol>
                <MDBCol>
                  <h1 className="font-weight-bold mb-3 p-0 black-text">
                    {article.view_count}
                  </h1>
                  <span className="black-text">views</span>
                </MDBCol>
              </MDBRow>
            </MDBCardText>
          </MDBCardBody>
        </MDBCard>
      </MDBCol>
    </MDBContainer>
  );
};
ReadingStats.propTypes = {
  article: propTypes.shape({})
};
ReadingStats.defaultProps = {
  article: {}
};
export default ReadingStats;
