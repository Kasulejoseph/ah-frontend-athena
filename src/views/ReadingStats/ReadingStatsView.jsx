import React, { Component } from "react";
import { connect } from "react-redux";
import propTypes from "prop-types";
import ReadingStats from "../../components/ReadingStats/ReadingStats";
import { getArticles } from "../../actions/articleActions/ArticleActions";

export class ReadingStatsView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      returned_articles: {}
    };
  }

  componentDidMount() {
    const { getArticles } = this.props;
    const username = window.localStorage.getItem("username");
    getArticles(`author=${username}`);
  }
  componentWillReceiveProps(nextProps) {
    const { articles } = nextProps;
    if (articles) {
      this.setState({ returned_articles: articles });
    } else {
      this.setState({ returned_articles: {} });
    }
  }
  render() {
    const { returned_articles } = this.state;
    if (JSON.stringify(returned_articles) !== "{}") {
      return returned_articles.map(article => {
        return <ReadingStats key={article.id} article={article} />;
      });
    } else {
      return <div />;
    }
  }
}
export const mapStateToProps = state => {
  return {
    articles: state.articles.articles
  };
};
ReadingStatsView.propTypes = {
  articles: propTypes.shape({}),
  getArticles: propTypes.func
};
ReadingStatsView.defaultProps = {
  articles: {},
  getArticles: () => {}
};

export default connect(
  mapStateToProps,
  { getArticles }
)(ReadingStatsView);
