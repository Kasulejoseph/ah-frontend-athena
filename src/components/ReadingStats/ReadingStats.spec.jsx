import React from "react";
import { shallow } from "enzyme";
import ReadingStats from "./ReadingStats";

let props = {
  article: {
    title: "",
    read_count: "",
    view_count: ""
  }
};

describe("Test if ReadingStats component mounts", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ReadingStats {...props} />);
  });
  it("Reading stats should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
