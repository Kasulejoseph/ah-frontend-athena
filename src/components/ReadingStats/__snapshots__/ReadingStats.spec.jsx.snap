// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if ReadingStats component mounts Reading stats should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReadingStats
    article={
      Object {
        "read_count": "",
        "title": "",
        "view_count": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <t
        className="flex flex-center"
        lg={null}
        md={null}
        sm={null}
        tag="div"
        xl={null}
        xs={null}
      >
        <t
          className="w-50"
          tag="div"
        >
          <t
            tag="div"
          >
            <t
              tag="p"
            >
              <t
                tag="div"
              >
                <t
                  lg={null}
                  md={null}
                  size="8"
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h3
                    className="font-weight-bold mb-3 p-0"
                  >
                    
                  </h3>
                </t>
                <t
                  lg={null}
                  md={null}
                  size="2"
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h1
                    className="font-weight-bold mb-3 p-0 black-text"
                  >
                    
                  </h1>
                  <span
                    className="black-text"
                  >
                    reads
                  </span>
                </t>
                <t
                  lg={null}
                  md={null}
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h1
                    className="font-weight-bold mb-3 p-0 black-text"
                  >
                    
                  </h1>
                  <span
                    className="black-text"
                  >
                    views
                  </span>
                </t>
              </t>
            </t>
          </t>
        </t>
      </t>,
      "fluid": false,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <t
          className="w-50"
          tag="div"
        >
          <t
            tag="div"
          >
            <t
              tag="p"
            >
              <t
                tag="div"
              >
                <t
                  lg={null}
                  md={null}
                  size="8"
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h3
                    className="font-weight-bold mb-3 p-0"
                  >
                    
                  </h3>
                </t>
                <t
                  lg={null}
                  md={null}
                  size="2"
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h1
                    className="font-weight-bold mb-3 p-0 black-text"
                  >
                    
                  </h1>
                  <span
                    className="black-text"
                  >
                    reads
                  </span>
                </t>
                <t
                  lg={null}
                  md={null}
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h1
                    className="font-weight-bold mb-3 p-0 black-text"
                  >
                    
                  </h1>
                  <span
                    className="black-text"
                  >
                    views
                  </span>
                </t>
              </t>
            </t>
          </t>
        </t>,
        "className": "flex flex-center",
        "lg": null,
        "md": null,
        "sm": null,
        "tag": "div",
        "xl": null,
        "xs": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <t
            tag="div"
          >
            <t
              tag="p"
            >
              <t
                tag="div"
              >
                <t
                  lg={null}
                  md={null}
                  size="8"
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h3
                    className="font-weight-bold mb-3 p-0"
                  >
                    
                  </h3>
                </t>
                <t
                  lg={null}
                  md={null}
                  size="2"
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h1
                    className="font-weight-bold mb-3 p-0 black-text"
                  >
                    
                  </h1>
                  <span
                    className="black-text"
                  >
                    reads
                  </span>
                </t>
                <t
                  lg={null}
                  md={null}
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h1
                    className="font-weight-bold mb-3 p-0 black-text"
                  >
                    
                  </h1>
                  <span
                    className="black-text"
                  >
                    views
                  </span>
                </t>
              </t>
            </t>
          </t>,
          "className": "w-50",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <t
              tag="p"
            >
              <t
                tag="div"
              >
                <t
                  lg={null}
                  md={null}
                  size="8"
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h3
                    className="font-weight-bold mb-3 p-0"
                  >
                    
                  </h3>
                </t>
                <t
                  lg={null}
                  md={null}
                  size="2"
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h1
                    className="font-weight-bold mb-3 p-0 black-text"
                  >
                    
                  </h1>
                  <span
                    className="black-text"
                  >
                    reads
                  </span>
                </t>
                <t
                  lg={null}
                  md={null}
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h1
                    className="font-weight-bold mb-3 p-0 black-text"
                  >
                    
                  </h1>
                  <span
                    className="black-text"
                  >
                    views
                  </span>
                </t>
              </t>
            </t>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <t
                tag="div"
              >
                <t
                  lg={null}
                  md={null}
                  size="8"
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h3
                    className="font-weight-bold mb-3 p-0"
                  >
                    
                  </h3>
                </t>
                <t
                  lg={null}
                  md={null}
                  size="2"
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h1
                    className="font-weight-bold mb-3 p-0 black-text"
                  >
                    
                  </h1>
                  <span
                    className="black-text"
                  >
                    reads
                  </span>
                </t>
                <t
                  lg={null}
                  md={null}
                  sm={null}
                  tag="div"
                  xl={null}
                  xs={null}
                >
                  <h1
                    className="font-weight-bold mb-3 p-0 black-text"
                  >
                    
                  </h1>
                  <span
                    className="black-text"
                  >
                    views
                  </span>
                </t>
              </t>,
              "tag": "p",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <t
                    lg={null}
                    md={null}
                    size="8"
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h3
                      className="font-weight-bold mb-3 p-0"
                    >
                      
                    </h3>
                  </t>,
                  <t
                    lg={null}
                    md={null}
                    size="2"
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h1
                      className="font-weight-bold mb-3 p-0 black-text"
                    >
                      
                    </h1>
                    <span
                      className="black-text"
                    >
                      reads
                    </span>
                  </t>,
                  <t
                    lg={null}
                    md={null}
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h1
                      className="font-weight-bold mb-3 p-0 black-text"
                    >
                      
                    </h1>
                    <span
                      className="black-text"
                    >
                      views
                    </span>
                  </t>,
                ],
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <h3
                      className="font-weight-bold mb-3 p-0"
                    >
                      
                    </h3>,
                    "lg": null,
                    "md": null,
                    "size": "8",
                    "sm": null,
                    "tag": "div",
                    "xl": null,
                    "xs": null,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                      "className": "font-weight-bold mb-3 p-0",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "h3",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <h1
                        className="font-weight-bold mb-3 p-0 black-text"
                      >
                        
                      </h1>,
                      <span
                        className="black-text"
                      >
                        reads
                      </span>,
                    ],
                    "lg": null,
                    "md": null,
                    "size": "2",
                    "sm": null,
                    "tag": "div",
                    "xl": null,
                    "xs": null,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "",
                        "className": "font-weight-bold mb-3 p-0 black-text",
                      },
                      "ref": null,
                      "rendered": "",
                      "type": "h1",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "reads",
                        "className": "black-text",
                      },
                      "ref": null,
                      "rendered": "reads",
                      "type": "span",
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <h1
                        className="font-weight-bold mb-3 p-0 black-text"
                      >
                        
                      </h1>,
                      <span
                        className="black-text"
                      >
                        views
                      </span>,
                    ],
                    "lg": null,
                    "md": null,
                    "sm": null,
                    "tag": "div",
                    "xl": null,
                    "xs": null,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "",
                        "className": "font-weight-bold mb-3 p-0 black-text",
                      },
                      "ref": null,
                      "rendered": "",
                      "type": "h1",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "views",
                        "className": "black-text",
                      },
                      "ref": null,
                      "rendered": "views",
                      "type": "span",
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <t
          className="flex flex-center"
          lg={null}
          md={null}
          sm={null}
          tag="div"
          xl={null}
          xs={null}
        >
          <t
            className="w-50"
            tag="div"
          >
            <t
              tag="div"
            >
              <t
                tag="p"
              >
                <t
                  tag="div"
                >
                  <t
                    lg={null}
                    md={null}
                    size="8"
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h3
                      className="font-weight-bold mb-3 p-0"
                    >
                      
                    </h3>
                  </t>
                  <t
                    lg={null}
                    md={null}
                    size="2"
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h1
                      className="font-weight-bold mb-3 p-0 black-text"
                    >
                      
                    </h1>
                    <span
                      className="black-text"
                    >
                      reads
                    </span>
                  </t>
                  <t
                    lg={null}
                    md={null}
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h1
                      className="font-weight-bold mb-3 p-0 black-text"
                    >
                      
                    </h1>
                    <span
                      className="black-text"
                    >
                      views
                    </span>
                  </t>
                </t>
              </t>
            </t>
          </t>
        </t>,
        "fluid": false,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <t
            className="w-50"
            tag="div"
          >
            <t
              tag="div"
            >
              <t
                tag="p"
              >
                <t
                  tag="div"
                >
                  <t
                    lg={null}
                    md={null}
                    size="8"
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h3
                      className="font-weight-bold mb-3 p-0"
                    >
                      
                    </h3>
                  </t>
                  <t
                    lg={null}
                    md={null}
                    size="2"
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h1
                      className="font-weight-bold mb-3 p-0 black-text"
                    >
                      
                    </h1>
                    <span
                      className="black-text"
                    >
                      reads
                    </span>
                  </t>
                  <t
                    lg={null}
                    md={null}
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h1
                      className="font-weight-bold mb-3 p-0 black-text"
                    >
                      
                    </h1>
                    <span
                      className="black-text"
                    >
                      views
                    </span>
                  </t>
                </t>
              </t>
            </t>
          </t>,
          "className": "flex flex-center",
          "lg": null,
          "md": null,
          "sm": null,
          "tag": "div",
          "xl": null,
          "xs": null,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <t
              tag="div"
            >
              <t
                tag="p"
              >
                <t
                  tag="div"
                >
                  <t
                    lg={null}
                    md={null}
                    size="8"
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h3
                      className="font-weight-bold mb-3 p-0"
                    >
                      
                    </h3>
                  </t>
                  <t
                    lg={null}
                    md={null}
                    size="2"
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h1
                      className="font-weight-bold mb-3 p-0 black-text"
                    >
                      
                    </h1>
                    <span
                      className="black-text"
                    >
                      reads
                    </span>
                  </t>
                  <t
                    lg={null}
                    md={null}
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h1
                      className="font-weight-bold mb-3 p-0 black-text"
                    >
                      
                    </h1>
                    <span
                      className="black-text"
                    >
                      views
                    </span>
                  </t>
                </t>
              </t>
            </t>,
            "className": "w-50",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <t
                tag="p"
              >
                <t
                  tag="div"
                >
                  <t
                    lg={null}
                    md={null}
                    size="8"
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h3
                      className="font-weight-bold mb-3 p-0"
                    >
                      
                    </h3>
                  </t>
                  <t
                    lg={null}
                    md={null}
                    size="2"
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h1
                      className="font-weight-bold mb-3 p-0 black-text"
                    >
                      
                    </h1>
                    <span
                      className="black-text"
                    >
                      reads
                    </span>
                  </t>
                  <t
                    lg={null}
                    md={null}
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h1
                      className="font-weight-bold mb-3 p-0 black-text"
                    >
                      
                    </h1>
                    <span
                      className="black-text"
                    >
                      views
                    </span>
                  </t>
                </t>
              </t>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <t
                  tag="div"
                >
                  <t
                    lg={null}
                    md={null}
                    size="8"
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h3
                      className="font-weight-bold mb-3 p-0"
                    >
                      
                    </h3>
                  </t>
                  <t
                    lg={null}
                    md={null}
                    size="2"
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h1
                      className="font-weight-bold mb-3 p-0 black-text"
                    >
                      
                    </h1>
                    <span
                      className="black-text"
                    >
                      reads
                    </span>
                  </t>
                  <t
                    lg={null}
                    md={null}
                    sm={null}
                    tag="div"
                    xl={null}
                    xs={null}
                  >
                    <h1
                      className="font-weight-bold mb-3 p-0 black-text"
                    >
                      
                    </h1>
                    <span
                      className="black-text"
                    >
                      views
                    </span>
                  </t>
                </t>,
                "tag": "p",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <t
                      lg={null}
                      md={null}
                      size="8"
                      sm={null}
                      tag="div"
                      xl={null}
                      xs={null}
                    >
                      <h3
                        className="font-weight-bold mb-3 p-0"
                      >
                        
                      </h3>
                    </t>,
                    <t
                      lg={null}
                      md={null}
                      size="2"
                      sm={null}
                      tag="div"
                      xl={null}
                      xs={null}
                    >
                      <h1
                        className="font-weight-bold mb-3 p-0 black-text"
                      >
                        
                      </h1>
                      <span
                        className="black-text"
                      >
                        reads
                      </span>
                    </t>,
                    <t
                      lg={null}
                      md={null}
                      sm={null}
                      tag="div"
                      xl={null}
                      xs={null}
                    >
                      <h1
                        className="font-weight-bold mb-3 p-0 black-text"
                      >
                        
                      </h1>
                      <span
                        className="black-text"
                      >
                        views
                      </span>
                    </t>,
                  ],
                  "tag": "div",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <h3
                        className="font-weight-bold mb-3 p-0"
                      >
                        
                      </h3>,
                      "lg": null,
                      "md": null,
                      "size": "8",
                      "sm": null,
                      "tag": "div",
                      "xl": null,
                      "xs": null,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "",
                        "className": "font-weight-bold mb-3 p-0",
                      },
                      "ref": null,
                      "rendered": "",
                      "type": "h3",
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <h1
                          className="font-weight-bold mb-3 p-0 black-text"
                        >
                          
                        </h1>,
                        <span
                          className="black-text"
                        >
                          reads
                        </span>,
                      ],
                      "lg": null,
                      "md": null,
                      "size": "2",
                      "sm": null,
                      "tag": "div",
                      "xl": null,
                      "xs": null,
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "",
                          "className": "font-weight-bold mb-3 p-0 black-text",
                        },
                        "ref": null,
                        "rendered": "",
                        "type": "h1",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "reads",
                          "className": "black-text",
                        },
                        "ref": null,
                        "rendered": "reads",
                        "type": "span",
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <h1
                          className="font-weight-bold mb-3 p-0 black-text"
                        >
                          
                        </h1>,
                        <span
                          className="black-text"
                        >
                          views
                        </span>,
                      ],
                      "lg": null,
                      "md": null,
                      "sm": null,
                      "tag": "div",
                      "xl": null,
                      "xs": null,
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "",
                          "className": "font-weight-bold mb-3 p-0 black-text",
                        },
                        "ref": null,
                        "rendered": "",
                        "type": "h1",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "views",
                          "className": "black-text",
                        },
                        "ref": null,
                        "rendered": "views",
                        "type": "span",
                      },
                    ],
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
